File: rust/cudf-sys/cpp/labeling.cpp
// SPDX-FileCopyrightText: Copyright (c) 2026, NVIDIA CORPORATION.
// SPDX-License-Identifier: Apache-2.0

#include "cudf-sys/src/labeling.rs.h"
#include "cudf-sys/src/lib.rs.h"
#include "cudf-sys/helpers.hpp"

#include <cudf/labeling/label_bins.hpp>

namespace cudf_sys {

std::unique_ptr<Column> label_bins_column(
    cudf::column_view const& col,
    cudf::column_view const& left_edges,
    int32_t left_inclusive,
    cudf::column_view const& right_edges,
    int32_t right_inclusive,
    std::size_t stream) {
  return COL(cudf::label_bins(col, left_edges,
      ENUM<cudf::inclusive>(left_inclusive),
      right_edges,
      ENUM<cudf::inclusive>(right_inclusive), S(stream)));
}

}  // namespace cudf_sys

File: rust/cudf-sys/src/labeling.rs
// SPDX-FileCopyrightText: Copyright (c) 2026, NVIDIA CORPORATION.
// SPDX-License-Identifier: Apache-2.0

#[cxx::bridge(namespace = "cudf_sys")]
pub mod ffi {
    unsafe extern "C++" {
        include!("cudf-sys/lib.hpp");

        // Shared types from the core bridge.
        type Column = crate::ffi::Column;
        #[namespace = "cudf"]
        type column_view = crate::ffi::column_view;

        // -- Label bins --

        /// Assigns bin labels to column values.
        fn label_bins_column(
            col: &column_view,
            left_edges: &column_view,
            left_inclusive: i32,
            right_edges: &column_view,
            right_inclusive: i32,
            stream: usize,
        ) -> Result<UniquePtr<Column>>;
    }
}
